examples/horizontal_convection.jl

```diff
@@ -0,0 +1,347 @@
+# boundary conditions on ``u`` and ``w`` at all boundaries.
+#
+# ### Non-dimensional control parameters
+#
+# The problem is characterized by three non-dimensional parameters. The first is the domain's
+# aspect ratio, ``A = L_x / H`` and the other two are the Rayleigh (``Ra``) and Prandtl (``Pr``)
+# numbers:
+#
+# ```math
+# Ra = \frac{L_x^3 b_*}{\nu \kappa} \, , \quad \text{and}\, \quad Pr = \frac{\nu}{\kappa} \, .
+# ```
+#
+# The Prandtl number expresses the ratio of momentum over heat diffusion; the Rayleigh number
+# is a measure of the relative importance of gravity over viscosity in the momentum equation.
+#
+# For a domain with a given extent, the nondimensional values of ``Ra`` and ``Pr`` uniquely
+# determine the viscosity and diffusivity, i.e.,
+# 
+# ```math
+# \nu = \sqrt{\frac{Pr b_* L_x^3}{Ra}} \quad \text{and} \quad \kappa = \sqrt{\frac{b_* L_x^3}{Pr Ra}} \, .
+# ```
+#
+# Now let's code these things up! We start off by importing `Oceananigans` and `Printf`.
+
+using Printf
+using Oceananigans
+
+# ## The grid
+#
+# We use a two-dimensional grid with an aspect ratio ``A = L_x / H = 2``.
+
+H = 1.0          # vertical domain extent
+Lx = 2H          # horizontal domain extent
+Nx, Nz = 128, 64 # horizontal, vertical resolution
+
+grid = RegularRectilinearGrid(size = (Nx, Nz),
+                                 x = (-Lx/2, Lx/2),
+                                 z = (-H, 0),
+                              halo = (3, 3),
+                          topology = (Bounded, Flat, Bounded))
+
+# Any attempts for `VerticallyStretchedRectilinearGrid` failed...
+# 
+# ```
+# σ = 0.2  # stretching factor
+# 
+# hyperbolically_spaced_faces(k) = H - H * (1 - tanh(σ * (k - 1) / Nz) / tanh(σ))
+# 
+# grid = VerticallyStretchedRectilinearGrid(size = (Nx, Nz),
+#                                           topology = (Bounded, Flat, Bounded),
+#                                           x=(-Lx/2, Lx/2),
+#                                           halo = (3, 3),
+#                                           z_faces = hyperbolically_spaced_faces)
+# 
+# # We plot vertical spacing versus depth to inspect the prescribed grid stretching:
+# using Plots
+# 
+# plot(grid.Δzᵃᵃᶜ[1:Nz], grid.zᵃᵃᶜ[1:Nz],
+#      marker = :circle,
+#      ylabel = "Depth",
+#      xlabel = "Vertical spacing",
+#      legend = nothing)
+# ```
+
+# ## Boundary conditions
+#
+# We impose a surface buoyancy boundary condition,
+#
+# ```math
+# b(x, z=0, t) = - b_* \cos (2π x / L_x) \, .
+# ```
+
+b★ = 1.0
+
+@inline bₛ(x, y, t) = - p.b★ * cos(2π * x / p.Lx)
+
+b_bcs = TracerBoundaryConditions(grid,
+                                 top = ValueBoundaryCondition(bₛ, parameters=(b★=b★, Lx=Lx)))
+
+# ## Turbulence closures
+#
+# We use isotropic viscosity and diffusivities, `ν` and `κ` whose values are obtain from the
+# prescribed ``Ra`` and ``Pr`` numbers. Here, we use ``Pr = 1`` and ``Ra = 10^8``:
+
+Pr = 1.0   # The Prandtl number
```

Comment:
I wonder if this code can handle higher Prandtl number, say 10?

Reply:
Why not?

(with appropriate Δx, Δt, ….)

Reply:
I presume that it can, given enough time.  I ask because people usually take `Pr` to be `1`, not because it's physical but because it's faster.  I suspect that changing `Pr` to `10` will slow things down, but if we run it on a `GPU`, then it might might run in a reasonable time.

Reply:
Agreed.